File: exercice2/java/src/test/java/fr/devoxx/devops/logs/sql/SparkSQL5Test.java
package fr.devoxx.devops.logs.sql;

import fr.devoxx.devops.logs.SparkTest;
import org.junit.Test;
import scala.Tuple4;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.closeTo;
import static org.hamcrest.Matchers.is;

public class SparkSQL5Test extends SparkTest {
    @Test
    public void test_spark() {
        SparkSQL5 test = new SparkSQL5();
        Tuple4<Long, Long, Double, Long> result = test.process(rdd(), sqlContext());
        assertThat(result._1(), is(6102L));
        assertThat(result._2(), is(40L));
        assertThat(result._3(), closeTo(90.044, 0.001));
        assertThat(result._4(), is(139L));
    }

}

File: exercice2/scala/src/test/scala/fr/devoxx/devops/logs/sql/SparkSQL5Test.scala
package fr.devoxx.devops.logs.sql

import fr.devoxx.devops.logs.{SharedSparkSQLContext, SharedSparkContext}
import org.scalatest.{FunSuite, Matchers}

class SparkSQL5Test extends FunSuite with SharedSparkSQLContext with Matchers {

  test("Statistiques sur la taille des requêtes") {
    val result = SparkSQL5(rdd, sqlContext).process
    result._1 shouldBe 6102
    result._2 shouldBe 40
    result._3 shouldBe 90.044 +- 1e-3
    result._4 shouldBe 139
  }
}
